src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Repository/FilterMetaRepository.cs

```diff
@@ -199,4 +218,64 @@ await optionTable
             )
             .ToList();
     }
+
+    private async Task<PublicIdMappings> CreatePublicIdMappings(
+        DataSetVersion dataSetVersion,
+        CancellationToken cancellationToken)
+    {
+        var mappings = await EntityFrameworkQueryableExtensions
+            .SingleOrDefaultAsync(publicDataDbContext
+                    .DataSetVersionMappings,
+                mapping => mapping.TargetDataSetVersionId == dataSetVersion.Id,
+                cancellationToken);
+
+        if (mappings is null)
+        {
+            return new PublicIdMappings();
+        }
+
+        var mappingsByFilter = mappings
+            .FilterMappingPlan
+            .Mappings
+            .ToDictionary(
+                keySelector: filter => filter.Key,
+                elementSelector: filter => filter
+                    .Value
+                    .OptionMappings
+                    .Values
+                    .Where(mapping => mapping.Type is MappingType.AutoMapped or MappingType.ManualMapped)
+                    .ToDictionary(
+                        keySelector: mapping => mapping.CandidateKey!,
+                        elementSelector: mapping => mapping.PublicId));
+
+        return new PublicIdMappings { Filters = mappingsByFilter };
+    }
+
+    private static string CreatePublicIdForFilterOptionMetaLink(
```

Comment:
We're not really creating anything so this might not be the best naming. Could also this to `GetFilterOptionMetaLinkPublicId`.

Reply:
Sounds good to me - renamed here and also the equivalent in LocationMetaRepository.  Also renamed `CreatePublicIdMappings` to `GetPublicIdMappings` in both.